com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/flowable/FlowableFacade.java

```diff
@@ -109,6 +113,29 @@ private boolean processHierarchyHasDeleteReason(HistoricProcessInstance historic
                        .anyMatch(this::processHierarchyHasDeleteReason);
     }
 
+    public boolean haveDeadLetterJobsAll(String processId) {
+        Set<String> processIds = getAllProcessExecutions(processId).stream()
+                                                                   .map(Execution::getProcessInstanceId)
+                                                                   .collect(Collectors.toSet());
+        Set<String> deadLetterJobIds = new HashSet<>();
+        for (String id : processIds) {
+            deadLetterJobIds.addAll(processEngine.getManagementService()
```

Comment:
1. Extract the processEngine.getManagementService()... into separate method.
2. Extract the processEngine.getManagementService().createDeadLetter..().processInstanceId...list() into separate variable